File: ios/UniPark/UniPark/domain/repositories/ScanRepository.swift
import Foundation

// TODO: implemented in data/repositories/
public protocol ScanRepository {
	func submitScan(
		passPayload: String,
		passSignature: String,
		direction: ScanDirection,
		lotId: UUID,
		idempotencyKey: String
	) async throws -> Scan

	func pendingScans() async throws -> [Scan]
	func syncPendingScans() async throws
}

File: android/app/src/main/java/com/unipark/android/domain/repositories/ScanRepository.kt
package com.unipark.android.domain.repositories

import com.unipark.android.domain.entities.Scan
import com.unipark.android.domain.entities.ScanDirection
import java.util.UUID

interface ScanRepository {
    suspend fun submitScan(
        qrPayload: String,
        direction: ScanDirection,
        lotId: UUID,
    ): Scan

    suspend fun syncPendingScans()
}
